src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5328,6 +5327,17 @@ private VisitArgumentResult VisitArgumentEvaluate(BoundExpression argument, RefK
                 case RefKind.In:
                     {
                         // Note: for lambda arguments, they will be converted in the context/state we saved for that argument
+                        if (conversion is { Kind: ConversionKind.ImplicitUserDefined })
```

Comment:
Why is this the right place to do this instead of within `VisitConversion`? VisitConversion has the operandType, it has the conversion, it can recreate the conversion from the argument state, etc.

Reply:
`VisitConversion` is used in many places. This change is specific to coercion of arguments.
Also, it would seem wrong for `VisitConversion` to generate a conversion.

Reply:
I had the impression that whenever we visit a user-defined conversion, we may need to re-infer the argument and return value nullability. Did I mix that up, and it only actually matters when the conversion is applied to a method argument?

Reply:
You're right. `VisitUserDefinedConversion` does classify conversions. Let me take another look at that. Thanks

---
In reply to: [522522284](https://github.com/dotnet/roslyn/pull/49279#discussion_r522522284) [](ancestors = 522522284)

Reply:
Explored this in commit 4, but I ended up reverting it. The logic is still specific to argument coercion, so it doesn't really help to put it in common method.

---
In reply to: [522528922](https://github.com/dotnet/roslyn/pull/49279#discussion_r522528922) [](ancestors = 522528922,522522284)

Reply:
It's not clear to me why this logic is specific to argument conversions. Won't we have the same requirements for assignments?

---
In reply to: [523153895](https://github.com/dotnet/roslyn/pull/49279#discussion_r523153895) [](ancestors = 523153895,522528922,522522284)